patroni/ha.py

```diff
@@ -1877,7 +1872,7 @@ def _on_shutdown(checkpoint_location: int) -> None:
                     # location, we can remove the leader key and allow them to start leader race.
 
                     # for a manual failover/switchover with a candidate, we should check the requested candidate only
-                    if self.is_failover_possible(self.get_failover_candidates(), cluster_lsn=checkpoint_location):
+                    if self.is_failover_possible(False, checkpoint_location):
```

Comment:
It seems like you changed the logic. In this place we have to do same checks as like normal failover happens:
1. if synchronous replication is enable - filter out non-sync nodes
2. if there is a failover.candidate - check that only the candidate received the checkpoint_location
3. and so on
 
```suggestion
                    if self.is_failover_possible(cluster_lsn=checkpoint_location):
```